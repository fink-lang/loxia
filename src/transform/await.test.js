import {fink2js} from '../testing';


test('compiles await', ()=> {
  expect(
    fink2js(`
      task1 = fn foo: - await foo

      task2 = fn foo: await (foo + 4)

      task3 = fn foo:
        bar = await foo()
        bar + 123

      a_gen = unfold curr=0:
        if:
          spam: await ni(curr)
          else : curr + 1

      await ni
    `)
  ).toMatchSnapshot();
});


