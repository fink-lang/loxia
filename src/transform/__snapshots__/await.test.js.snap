// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles await 1`] = `
"const task1 = async foo => -(await foo);

const task2 = async foo => await (foo + 4);

const task3 = async foo => {
  const bar = await foo();
  return bar + 123;
};

const a_gen = async function* unfold(curr = 0) {
  while (true) {
    let _do_result;

    if (spam) {
      _do_result = await ni(curr);
    } else {
      _do_result = curr + 1;
    }

    const $result1 = _do_result;
    _do_result = undefined;
    if ($result1 === stop) return;
    if ($result1 !== skip) yield $result1;
  }
}();

await ni;"
`;
